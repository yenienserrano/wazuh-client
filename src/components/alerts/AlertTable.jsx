import React from "react";

export const AlertTable = ({ alert, alertDetail }) => {
  return (
    <tr
      className="pointer animate__animated animate__fadeIn"
      onClick={() => alertDetail(alert)}
    >
      <td>{alert._id}</td>
      <td>{alert._source.agent.name}</td>
      <td>{alert._source.rule.id}</td>
      <td>{alert._source.syscheck.event}</td>
    </tr>
  );
};
