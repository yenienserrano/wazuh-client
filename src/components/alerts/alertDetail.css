.containerCss {
  position: absolute;
  right: 0;
  top: 0;
  bottom: 0;
  width: 500px;
}
