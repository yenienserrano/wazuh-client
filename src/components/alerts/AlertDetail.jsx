import React from "react";
import "./alertDetail.css";

export const AlertDetail = ({ setShow, alert }) => {
  return (
    <div className="containerCss text-white bg-dark d-flex flex-column align-items-center justify-content-center">
      <div>
        <h1>Alert detail</h1>
        <div>
          <p>
            <b>Id:</b> {alert._id}
          </p>
          <p>
            <b>Index:</b> {alert._index}
          </p>
          <p>
            <b>Type:</b> {alert._type}
          </p>
        </div>
        <h3>Alert rule detail</h3>
        <div>
          <p>
            <b>Id:</b> {alert._source.rule.id}
          </p>
          <p>
            <b>Description:</b> {alert._source.rule.description}
          </p>
          <p>
            <b>Level:</b> {alert._source.rule.level}
          </p>
        </div>
        <h3>Alert agents detail</h3>
        <div>
          <p>
            <b>Id:</b> {alert._source.agent.id}
          </p>
          <p>
            <b>Name:</b> {alert._source.agent.name}
          </p>
          <p>
            <b>Ip:</b> {alert._source.agent.ip}
          </p>
        </div>
        <h3>Other</h3>
        <div>
          <p>
            <b>Manager:</b> {alert._source.manager.name}
          </p>
          <p>
            <b>cluster:</b> {alert._source.cluster.name}
          </p>
          <p>
            <b>Event:</b> {alert._source.syscheck.event}
          </p>
        </div>
      </div>
      <div>
        <button
          className="btn btn-outline-light"
          onClick={() => setShow(false)}
        >
          Cerrar
        </button>
      </div>
    </div>
  );
};
