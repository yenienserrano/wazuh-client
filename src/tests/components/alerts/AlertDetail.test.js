import React from "react";
import { shallow } from "enzyme";
import { AlertDetail } from "../../../components/alerts/AlertDetail";

describe("Test in <AlertDetail />", () => {
  const data = {
    _index: "wazuh-alerts-4.x-sample-security",
    _type: "_doc",
    _id: "23Gmr3QB4YtWQbLv1inX",
    _score: 1,

    _source: {
      "@sampledata": true,
      timestamp: "2020-09-14T18:30:45.569+0000",

      rule: {
        firedtimes: 1,
        mail: false,
        level: 5,

        pci_dss: ["11.5"],

        hipaa: ["164.312.c.1", "164.312.c.2"],
        description: "File added to the system.",

        groups: ["ossec", "syscheck"],
        id: "554",

        nist_800_53: ["SI.7"],

        gpg13: ["4.11"],

        gdpr: ["II_5.1.f"],
      },

      agent: {
        id: "004",
        name: "Ubuntu",
        ip: "47.204.15.21",
      },

      manager: {
        name: "wazuh-master",
      },

      cluster: {
        name: "wazuh",
      },
      id: "1580123327.49031",

      predecoder: {},

      decoder: {},

      data: {},
      location: "",

      syscheck: {
        event: "added",
        path: "/run/utmp",
        uname_after: "NETWORK Service",
        gname_after: "root",
        mtime_after: "2020-09-14T19:40:09.863Z",
        size_after: 18,
        uid_after: "S-1-5-18",
        gid_after: "190",
        perm_after: "rw-r--r--",
        inode_after: 41722,
      },
    },
  };

  const wrapper = shallow(<AlertDetail alert={data} />);

  test("should display the component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should show an h1 with the text "<b>Id:</b>23Gmr3QB4YtWQbLv1inX" ', () => {
    const h1 = wrapper.find("h1");
    expect(h1.text()).toBe("Alert detail");
  });
});
