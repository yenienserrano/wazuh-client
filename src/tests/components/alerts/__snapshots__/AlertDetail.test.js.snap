// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <AlertDetail /> should display the component correctly 1`] = `
<div
  className="containerCss text-white bg-dark d-flex flex-column align-items-center justify-content-center"
>
  <div>
    <h1>
      Alert detail
    </h1>
    <div>
      <p>
        <b>
          Id:
        </b>
         
        23Gmr3QB4YtWQbLv1inX
      </p>
      <p>
        <b>
          Index:
        </b>
         
        wazuh-alerts-4.x-sample-security
      </p>
      <p>
        <b>
          Type:
        </b>
         
        _doc
      </p>
    </div>
    <h3>
      Alert rule detail
    </h3>
    <div>
      <p>
        <b>
          Id:
        </b>
         
        554
      </p>
      <p>
        <b>
          Description:
        </b>
         
        File added to the system.
      </p>
      <p>
        <b>
          Level:
        </b>
         
        5
      </p>
    </div>
    <h3>
      Alert agents detail
    </h3>
    <div>
      <p>
        <b>
          Id:
        </b>
         
        004
      </p>
      <p>
        <b>
          Name:
        </b>
         
        Ubuntu
      </p>
      <p>
        <b>
          Ip:
        </b>
         
        47.204.15.21
      </p>
    </div>
    <h3>
      Other
    </h3>
    <div>
      <p>
        <b>
          Manager:
        </b>
         
        wazuh-master
      </p>
      <p>
        <b>
          cluster:
        </b>
         
        wazuh
      </p>
      <p>
        <b>
          Event:
        </b>
         
        added
      </p>
    </div>
  </div>
  <div>
    <button
      className="btn btn-outline-light"
      onClick={[Function]}
    >
      Cerrar
    </button>
  </div>
</div>
`;
